### PR TITLE
Create python-publish.yml

### PR DESCRIPTION
**Pull request by lucpeterson**

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+    - name: Build package
+      run: python -m build
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
```
Comment (koning) [inline .github/workflows/python-publish.yml:33]: That's a very specific commit, is there a story behind that?

**Comment by lucpeterson** [comment]
This was "auto-generated" when I clicked the GitHub action button for this. It looks like this hash is tied to v.1.4.2 (Feb 2021). They just added a v 1.5.0 a few days ago we could change it to, but this is the script that GitHub writ large creates when you use it to add this action.

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (lucpeterson) [reply .github/workflows/python-publish.yml:33]: This was "auto-generated" when I clicked the GitHub action button for this. It looks like this hash is tied to v.1.4.2 (Feb 2021). They just added a v 1.5.0 a few days ago we could change it to, but this is the script that GitHub writ large creates when you use it to add this action.

Comment (koning) [reply .github/workflows/python-publish.yml:33]: OK, looks good to me.

**Review by koning — APPROVED** [review]

**Pull request merged**